Framework/Algorithms/src/FindEPP.cpp

```diff
@@ -0,0 +1,196 @@
+#include "MantidAlgorithms/FindEPP.h"
+#include "MantidAPI/TableRow.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidKernel/make_unique.h"
+
+#include <cmath>
+#include <sstream>
+
+namespace Mantid {
+namespace Algorithms {
+
+using namespace Mantid::Kernel;
+using namespace Mantid::API;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(FindEPP)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string FindEPP::name() const { return "FindEPP"; }
+
+/// Algorithm's version for identification. @see Algorithm::version
+int FindEPP::version() const { return 2; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string FindEPP::category() const {
+  return "Workflow\\MLZ\\TOFTOF;Utility";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string FindEPP::summary() const {
+  return "Performs Gaussian fits over each spectrum to find the Elastic Peak "
+         "Position (EPP).";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void FindEPP::init() {
+  declareProperty(Kernel::make_unique<WorkspaceProperty<API::MatrixWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "An input workspace.");
+  declareProperty(Kernel::make_unique<WorkspaceProperty<API::ITableWorkspace>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "An output workspace.");
+}
+
+//----------------------------------------------------------------------------------------------
+/** Execute the algorithm.
+ */
+void FindEPP::exec() {
+  m_inWS = getProperty("InputWorkspace");
+
+  initWorkspace();
+
+  int64_t numberspectra = static_cast<int64_t>(m_inWS->getNumberHistograms());
+
+  // Loop over spectra
+  PARALLEL_FOR_IF(threadSafe(*m_inWS, *m_outWS))
+  for (int64_t index = 0; index < numberspectra; ++index) {
+    PARALLEL_START_INTERUPT_REGION
+    fitGaussian(index);
+    PARALLEL_END_INTERUPT_REGION
+  }
+  PARALLEL_CHECK_INTERUPT_REGION
+
+  setProperty("OutputWorkspace", m_outWS);
+}
+
+/* Call Fit as child algorithm for each spectra
+ * @param index : the workspace index
+ */
+void FindEPP::fitGaussian(int64_t index) {
+  size_t spectrum = static_cast<size_t>(index);
+  const auto &x = m_inWS->x(spectrum).rawData();
+  const auto &y = m_inWS->y(spectrum).rawData();
+  const auto &e = m_inWS->e(spectrum).rawData();
+
```

Comment:
I think some comment should be added between here and L104 to explain why the algorithm is doing these operations. I had to go and read the source code for the .py version to understand this one.